### DISCUSSION TITLE
OS detection error on raspberry pi 3B

### DISCUSSION THREAD

**Discussion by jjpd19** [pivpn/pivpn#1397] [Feature requests]
As the topic says, I am getting this error when starting the installation. I have installed osmc on it and a plex server.
![imagen](https://user-images.githubusercontent.com/37415796/72510349-93b37080-3849-11ea-9a52-af233980ce1f.png)

**Comment by orazioedoardo** [comment]
Output of `cat /etc/os-release`?

**Comment by coolapso** [comment]
That seems veeery weird.. are you sure you using Raspbian? 
check what @orazioedoardo mentioned.

**Comment by jjpd19** [comment]
I know it is very weird but... I am writing from my tablet so I am copying the answer of the execution of the command

Linux osmc 4.19.55-6-osmc #1 SMP PREEMPT Sun Nov 3 22:15:28 UTC 2019 armv7l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Thu Jan 16 11:44:31 2020 from 192.168.1.155
osmc@osmc:~$ cat /etc/os-release
PRETTY_NAME="Open Source Media Center"
NAME="OSMC"
VERSION="November 2019"
VERSION_ID="2019.11-1"
ID=osmc
ID_LIKE=debian

ANSI_COLOR="1;31"
HOME_URL="https://www.osmc.tv"
SUPPORT_URL="https://www.osmc.tv"
BUG_REPORT_URL="https://www.osmc.tv"
osmc@osmc:~$

**Comment by coolapso** [comment]
well .. the script is right! you are not using a supported os!! that's OSMC own image!

I think you can use Raspbian and install osmc on top of it, if you want to use pivpn thats my advice. 

We might try to support OSMC in the future but that's not a thing for now!

**Comment by jjpd19** [comment]
Ok. Thank you. I guessed that osmc was raspbian.

**Comment by coolapso** [comment]
Adding tags for refference and test and ivestigate how much trouble is to support OSMC

**Comment by 29antonioac** [comment]
I have PiVPN installed in OSMC without any issues. I just had to install `lsb-release` to make it compatible.

I need to fix that sometimes OpenVPN is executed before the network so tun0 doesn't get any IP, so I need to restart it manually. I have overriden these variables in the systemd service unit, but sometimes doesn't work as expected.

```
# /etc/systemd/system/openvpn.service.d/override.conf

After=network-online.target
Wants=network.target network-online.target
``` 

Hope to be useful!

**Comment by coolapso** [comment]
that's a nice piece of info there! thank you!

**Comment by MembershipExcel** [comment]
Could you add Linux mint in supported OS ? Of adding an alert saying it not officialy suported ? It debian based..

**Comment by AnayaBit** [comment]
I have the same problem with another OS based on debian :( 
Linux pihole-server 4.19.0-9-686 #1 SMP Debian 4.19.118-2+deb10u1 (2020-06-07) i686 GNU/Linux
pihole@pihole-server:~$ cat /etc/os-release
PRETTY_NAME="SparkyLinux 5.11 (Nibiru)"
NAME="SparkyLinux"
VERSION_ID="5.11"
VERSION="5.11 (Nibiru)"
ID=sparky
ID_LIKE=debian
HOME_URL="https://sparkylinux.org/"
SUPPORT_URL="https://sparkylinux.org/forum/"
BUG_REPORT_URL="https://sourceforge.net/p/sparkylinux/tickets"
